### DISCUSSION TITLE
For managed Supabase, are `auth` etc. schemas "private"?

### DISCUSSION THREAD

**Discussion by quicksnap** [supabase/supabase#1565] [Questions] [answered]
I figured I would change my `auth.users(email)` to type `CITEXT`. But, that got me to thinking.. as Supabase evolves, how do managed instances deal with any changes needed to schemas that work with the client API, such as `auth`, `storage`, etc.?

I'm sure it's quite easy for me to break things in these schemas.. but that's on me :). But, I'd hate to do something that ruins forward compatibility with supabase-js.

So... how's that all work? =P

**Comment by awalias** [comment, marked as answer]
great questions @quicksnap !

in general we suggest not tampering with the auth and storage schemas 😆 . We could potentially hide them away from the `postgres` user but we felt that we don't want things to be `magic`. If you want to extend the functionality the recommended approach is something like [adding a `public.users` table that inserts users using a postgres trigger when people sign up](https://supabase.io/docs/guides/auth#create-a-publicusers-table).

If you have specific things that could be improved (e.g. changing auth.users(email) to type CITEXT) - you might consider adding them to `https://github.com/supabase/gotrue` so they can potentially be rolled out to everyone

for breaking changes (where we make changes to gotrue for example, we would indeed run a schema migration to make sure it's in sync with the server version)
